src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Portal/New-ArmDashboardsQuickPulseButtonSmall.ps1

```diff
@@ -0,0 +1,43 @@
+function New-ArmDashboardsQuickPulseButtonSmall {
+  [CmdletBinding(SupportsShouldProcess = $true)]
+  [OutputType("DashboardPart")]
+  Param(
+    [PSTypeName("ApplicationInsights")]
+    [Parameter(Mandatory)]
+    $ApplicationInsights,
+    [string]
+    [Parameter(Mandatory)]
+    $SubscriptionId,
+    [string]
+    [Parameter(Mandatory)]
+    $ResourceGroupName
+  )
+
+  If ($PSCmdlet.ShouldProcess("Adding QuickPulseButtonSmallPart to Dashboards")) {
+    $ApplicationInsightsResourceName = $ApplicationInsights.Name
+    $ApplicationInsightsResourceId = $ApplicationInsights._ResourceId
+    $QuickPulse = [PSCustomObject][ordered]@{
+      PSTypeName = "DashboardPart"
+      position   = @{ }
+      metadata   = @{
+        inputs = @(@{
```

Comment:
It seems that this code is reused at a lot of places 
```
@{
  name  = 'ComponentId'
  value = @{
    Name = $ApplicationInsightsResourceName
    SubscriptionId = $SubscriptionId
    ResourceGroup  = $ResourceGroupName
  }
}
```

Should we extract it?